code/modules/client/client_procs.dm

```diff
@@ -333,8 +335,14 @@
 		alert(src, "You are using a byond build which is not supported by this server. Please use a build version of atleast [config.minimum_client_build].", "Incorrect build", "OK")
 		qdel(src)
 		return
+	var/show_update_prompt = FALSE
 	if(byond_version < SUGGESTED_CLIENT_VERSION) // Update is suggested, but not required.
-		to_chat(src,"<span class='userdanger'>Your BYOND client (v: [byond_version]) is out of date. This can cause glitches. We highly suggest you download the latest client from http://www.byond.com/ before playing. </span>")
+		show_update_prompt = TRUE
+	else if(byond_version == SUGGESTED_CLIENT_VERSION && byond_build < SUGGESTED_CLIENT_BUILD)
+		show_update_prompt = TRUE
+	if(show_update_prompt)
+		to_chat(src,"<span class='userdanger'>Your BYOND client (v: [byond_version].[byond_build]) is out of date. This can cause glitches. We highly suggest you download the latest client from http://www.byond.com/ before playing. </span>")
```

Comment:
These messages won't even be seen because goonchat doesn't even start for another ~100 lines or so.

This entire system annoys me a bit (not your code, the whole version checking thing), especially since it's sending alerts to the user and then promptly qdel-ing the client before they even get a chance to see the alerts at all.

Reply:
I tested it, and saw the messages.

Reply:
Confimed. Its `alert()` and `to_chat()` which differ.